apis/v1/gateway_types.go

```diff
@@ -672,6 +672,7 @@ type GatewayAddress struct {
 	//
 	// Examples: `1.2.3.4`, `128::1`, `my-ip-address`.
 	//
+	// +optional
```

Comment:
This needs some more explanation:

* what behavior is expected if an implementation supports an empty `Value` here?
* what behavior is expected if an implementation *does not* support an empty `Value` here?

This should then guide you to adding some conformance tests for implementations to test against.

Reply:
+1 to @youngnick's comments, a couple suggestions as a starting point:

1. If an implementation does support an empty value, we expect them to assign an address matching the type that has been requested with a value of the implementation's choice.
2. If an implementation does _not_ support an empty value, we expect them to set the `Programmed` condition in status to `False` with a reason of `AddressNotAssigned`. Here's what that reason means as a reference:

https://github.com/kubernetes-sigs/gateway-api/blob/d936258b0ed256e2d4eaa31da01fa703f0e954b0/apis/v1/gateway_types.go#L874-L886

Reply:
I think the extra doc requirement here got missed. I'd still like to see some description of what should happen in both cases _on this field_, if it's supported and not.

Reply:
Thanks for catching this @youngnick! @EyalPazz any chance you can squeeze this in in the next day or two? Really hoping we can fit this into v1.3 if possible, and we'll need to cut a RC pretty soon.